File: src/main/java/openhds/service/MembershipService.java
package openhds.service;

import openhds.domain.Membership;
import openhds.domain.Request;
import org.springframework.stereotype.Service;

@Service
public class MembershipService  extends HttpService {
    @Override
    public String createEntity(Request request) {
        return create("/memberships", request, Membership.class);
    }
}

File: src/ohds/membership_service.clj
(ns ohds.membership-service
  (:require [ohds.service :refer [create-entity]]))

(defn model->request
  "Transform frontend model to rest model"
  [{:keys [individual socialGroup startType startDate
           collectionDateTime collectedByUuid]}]
  {:membership  {:individual {:uuid individual}
                 :socialGroup {:uuid socialGroup}
                 :startType startType
                 :startDate startDate
                 :collectionDateTime collectionDateTime}
   :collectedByUuid collectedByUuid})

(defn create-membership
  [request]
  (->> (model->request request)
       (create-entity "/memberships")))
